cpp/ql/lib/semmle/code/cpp/dataflow/new/DataFlow.qll

```diff
@@ -0,0 +1,30 @@
+/**
+ * Provides a library for local (intra-procedural) and global (inter-procedural)
+ * data flow analysis: deciding whether data can flow from a _source_ to a
+ * _sink_. This library differs from the one in `semmle.code.cpp.dataflow` in that
+ * this library uses the IR (Intermediate Representation) library, which provides
+ * a more precise semantic representation of the program, whereas the other dataflow
+ * library uses the more syntax-oriented ASTs. This library should provide more accurate
+ * results than the AST-based library in most scenarios.
+ *
+ * Unless configured otherwise, _flow_ means that the exact value of
+ * the source may reach the sink. We do not track flow across pointer
+ * dereferences or array indexing.
```

Comment:
I'm not actually sure what this sentence means. Does:
> We do not track flow across pointer dereferences or array indexing.

mean that there's not a general dataflow step from `p` to `*p`? Or does it mean that there's not dataflow in an example like:
```cpp
int x = source();
int* p = &x;
sink(*p);
```
?

If we take the sentence to mean the former, then the statement is true. However, if the intended meaning is the latter, then it's false (since we use the indirect dataflow nodes to carry flow through indirections).

Reply:
This comes directly from `semmle.code.cpp.ir.dataflow`, and I was just a confused about this as you. Maybe we should just drop that text in both places?

Reply:
Yeah, I think that makes sense. Before use-use the statement was correct under both interpretations, and we never updated the QLDoc on the use-use flow branch. I think now is a good time to do so 😄.

Reply:
Should be fixed in b9200c92d5524e6faf6da3479b16354cdfd49ce1